guiTests/guiTest.py

```diff
@@ -101,67 +103,58 @@ def wait_for_text(time, xpath, text):
-driver.find_element_by_id("passphrase").send_keys(PASSWORD)
-driver.find_element_by_id("repeat-passphrase").send_keys(PASSWORD)
-driver.find_element_by_id("nextStep1").click()
+wait_for_load(10, By.ID, "name").send_keys(BACKUP_NAME)
+wait_for_load(10, By.ID, "passphrase").send_keys(PASSWORD)
+wait_for_load(10, By.ID, "repeat-passphrase").send_keys(PASSWORD)
+wait_for_load(10, By.ID, "nextStep1").click()
 
 # Add new backup - Destination page
-time.sleep(1)
-driver.find_element_by_link_text("Manually type path").click()
-driver.find_element_by_id("file_path").send_keys(DESTINATION_FOLDER)
-driver.find_element_by_id("nextStep2").click()
+wait_for_load(10, By.LINK_TEXT, "Manually type path").click()
+wait_for_load(10, By.ID, "file_path").send_keys(DESTINATION_FOLDER)
+wait_for_load(10, By.ID, "nextStep2").click()
 
 # Add new backup - Source Data page
-time.sleep(1)
-driver.find_element_by_id("sourcePath").send_keys(os.path.abspath(SOURCE_FOLDER) + os.sep)
-driver.find_element_by_id("sourceFolderPathAdd").click()
-driver.find_element_by_id("nextStep3").click()
+wait_for_load(10, By.ID, "sourcePath").send_keys(os.path.abspath(SOURCE_FOLDER) + os.sep)
+wait_for_load(10, By.ID, "sourceFolderPathAdd").click()
+wait_for_load(10, By.ID, "nextStep3").click()
 
 # Add new backup - Schedule page
-time.sleep(1)
-useScheduleRun = driver.find_element_by_id("useScheduleRun")
+useScheduleRun = wait_for_load(10, By.ID, "useScheduleRun")
 if useScheduleRun.is_selected():
     useScheduleRun.click()
-driver.find_element_by_id("nextStep4").click()
+wait_for_load(10, By.ID, "nextStep4").click()
 
 # Add new backup - Options page
-time.sleep(1)
-driver.find_element_by_id("save").click()
+wait_for_load(10, By.ID, "save").click()
 
 # Run the backup job and wait for finish
-time.sleep(1)
-driver.find_element_by_link_text(BACKUP_NAME).click()
+wait_for_load(10, By.LINK_TEXT, BACKUP_NAME).click()
 [n for n in driver.find_elements_by_xpath("//dl[@class='taskmenu']/dd/p/span[contains(text(),'Run now')]") if n.is_displayed()][0].click()
 wait_for_text(60, "//div[@class='task ng-scope']/dl[2]/dd[1]", "(took ")
 
 # Restore
 if len([n for n in driver.find_elements_by_xpath(u"//span[contains(text(),'Restore files \u2026')]") if n.is_displayed()]) == 0:
-    driver.find_element_by_link_text(BACKUP_NAME).click()
+    wait_for_load(10, By.LINK_TEXT, BACKUP_NAME).click()
 
 [n for n in driver.find_elements_by_xpath(u"//span[contains(text(),'Restore files \u2026')]") if n.is_displayed()][0].click()
-driver.find_element_by_xpath("//span[contains(text(),'" + SOURCE_FOLDER + "')]")  # wait for filelist
+wait_for_load(10, By.XPATH, "//span[contains(text(),'" + SOURCE_FOLDER + "')]")  # wait for filelist
 time.sleep(1)
-driver.find_element_by_xpath("//restore-file-picker/ul/li/div/a[2]").click()  # select root folder checkbox
+wait_for_load(10, By.XPATH, "/html/body/div[1]/div[2]/div[2]/div/form/div[1]/div[3]/restore-file-picker/ul/li/div/a[2]").click()  # select root folder checkbox
```

Comment:
Just curious, is there a reason why the provided path needed to be changed?

Reply:
I changed it while trying to get the checkbox selection working, but it was not the reason for the problem.

Reply:
Thanks!